pkg/sessionctx/stmtctx/stmtctx.go

```diff
@@ -1255,9 +1242,13 @@ func (sc *StatementContext) SetFlagsFromPBFlag(flags uint64) {
 	sc.TruncateAsWarning = (flags & model.FlagTruncateAsWarning) > 0
 	sc.InInsertStmt = (flags & model.FlagInInsertStmt) > 0
 	sc.InSelectStmt = (flags & model.FlagInSelectStmt) > 0
+	sc.InDeleteStmt = (flags & model.FlagInUpdateOrDeleteStmt) > 0
 	sc.OverflowAsWarning = (flags & model.FlagOverflowAsWarning) > 0
 	sc.IgnoreZeroInDate = (flags & model.FlagIgnoreZeroInDate) > 0
 	sc.DividedByZeroAsWarning = (flags & model.FlagDividedByZeroAsWarning) > 0
+	sc.SetTypeFlags(typectx.StrictFlags.
```

Comment:
SetFlagsFromPBFlag looks the same with flagsToStatementContext in the mock store except it did not set InDeleteStmt (I think we missed it before). So I replace the implement in flagsToStatementContext by calling SetFlagsFromPBFlag directly